Formula/soplex.rb

```diff
@@ -0,0 +1,44 @@
+class Soplex < Formula
+  desc "Optimization package for solving linear programming problems (LPs)"
+  homepage "https://soplex.zib.de/"
+  url "https://soplex.zib.de/download/release/soplex-6.0.3.tgz"
+  sha256 "357db78d18d9f1b81a4fe65cf0182475d2a527a6ad9030ac2f5ef7c077d6764c"
+  license "Apache-2.0"
+
+  depends_on "cmake" => :build
+  depends_on "boost"
+  depends_on "gmp"
+  depends_on "tbb"
+  uses_from_macos "zlib"
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+    cd "build" do
+      system "make"
+      system "make", "install"
+    end
+    pkgshare.install "src/example.cpp"
+  end
+
+  test do
+    (testpath/"test.lp").write <<~EOS
+      Maximize
+       obj: x1 + 2 x2 + 3 x3 + x4
+      Subject To
+       c1: - x1 + x2 + x3 + 10 x4 <= 20
+       c2: x1 - 3 x2 + x3 <= 30
+       c3: x2 - 3.5 x4 = 0
+      Bounds
+       0 <= x1 <= 40
+       2 <= x4 <= 3
+      General
+       x4
+      End
+    EOS
+    assert_match "problem is solved [optimal]", shell_output("#{bin}/soplex test.lp")
+
+    system ENV.cxx, pkgshare/"example.cpp", "-std=c++14", "-L#{lib}", "-I#{include}",
```

Comment:
can we use c++17?

Reply:
The project uses C++14, so we should use that.

Reply:
yeah, make sense!